wallet/test/backend_test.go

```diff
@@ -0,0 +1,96 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"io"
+	"math/rand"
+	"perun.network/perun-cardano-backend/wallet"
+	"testing"
+)
+
+func TestBackend_NewAddress(t *testing.T) {
+	seed := SetSeed()
+	r := NewMockRemote()
+	uut := wallet.MakeRemoteBackend(r)
+	actualAddress := uut.NewAddress()
+	_, ok := actualAddress.(*wallet.PubKey)
+	require.Truef(t, ok, "NewAddress() does not return a PubKey, test-seed: %d", seed)
+}
+
+func TestBackend_DecodeSig(t *testing.T) {
+	seed := SetSeed()
+	r := NewMockRemote()
+	uut := wallet.MakeRemoteBackend(r)
+
+	readerExact := bytes.NewReader(r.MockSignature)
+	actualSig, err := uut.DecodeSig(readerExact)
+	require.NoErrorf(t, err, "received an error when decoding signature, test-seed: %d", seed)
+	require.Equalf(t, r.MockSignature, actualSig, "decoded signature is incorrect, test-seed: %d", seed)
+
+	randomBytes := make([]byte, rand.Intn(129))
```

Comment:
Either completely random or give `129` a meaningful name and extract it into a function local constant. Why specifically `129` anyway?

Reply:
Because I want to test data up to 128 bytes long, `rand.Intn` has an exclusive upper-bound and 128 bytes seems less specific than 127 :D